Consistently use wxFont.. types instead of int; completes SVN r56664 (FM) which
unfortunately broke compilation of src/{dfb,mgl}/font.cpp

Signed-off-by: Anders Larsen <al@alarsen.net>
---
 include/wx/private/fontmgr.h |   18 +++++++++---------
 src/common/fontmgrcmn.cpp    |   26 +++++++++++---------------
 2 files changed, 20 insertions(+), 24 deletions(-)

Index: b/include/wx/private/fontmgr.h
===================================================================
--- a/include/wx/private/fontmgr.h
+++ b/include/wx/private/fontmgr.h
@@ -205,9 +205,9 @@ class wxFontMgrFontRefData : public wxGD
 {
 public:
     wxFontMgrFontRefData(int size = wxDEFAULT,
-                  int family = wxDEFAULT,
-                  int style = wxDEFAULT,
-                  int weight = wxDEFAULT,
+                  wxFontFamily family = wxFONTFAMILY_DEFAULT,
+                  wxFontStyle style = wxFONTSTYLE_NORMAL,
+                  wxFontWeight weight = wxFONTWEIGHT_NORMAL,
                   bool underlined = false,
                   const wxString& faceName = wxEmptyString,
                   wxFontEncoding encoding = wxFONTENCODING_DEFAULT);
@@ -223,16 +223,16 @@ public:
 
     int GetPointSize() const { return m_info.pointSize; }
     wxString GetFaceName() const { return m_info.faceName; }
-    int GetFamily() const { return m_info.family; }
-    int GetStyle() const { return m_info.style; }
-    int GetWeight() const { return m_info.weight; }
+    wxFontFamily GetFamily() const { return m_info.family; }
+    wxFontStyle GetStyle() const { return m_info.style; }
+    wxFontWeight GetWeight() const { return m_info.weight; }
     bool GetUnderlined() const { return m_info.underlined; }
     wxFontEncoding GetEncoding() const { return m_info.encoding; }
 
     void SetPointSize(int pointSize);
-    void SetFamily(int family);
-    void SetStyle(int style);
-    void SetWeight(int weight);
+    void SetFamily(wxFontFamily family);
+    void SetStyle(wxFontStyle style);
+    void SetWeight(wxFontWeight weight);
     void SetFaceName(const wxString& faceName);
     void SetUnderlined(bool underlined);
     void SetEncoding(wxFontEncoding encoding);
Index: b/src/common/fontmgrcmn.cpp
===================================================================
--- a/src/common/fontmgrcmn.cpp
+++ b/src/common/fontmgrcmn.cpp
@@ -225,19 +225,15 @@ void wxFontsManagerBase::AddBundle(wxFon
 // ----------------------------------------------------------------------------
 
 wxFontMgrFontRefData::wxFontMgrFontRefData(int size,
-                                           int family,
-                                           int style,
-                                           int weight,
+                                           wxFontFamily family,
+                                           wxFontStyle style,
+                                           wxFontWeight weight,
                                            bool underlined,
                                            const wxString& faceName,
                                            wxFontEncoding encoding)
 {
-    if ( family == wxDEFAULT )
-        family = wxSWISS;
-    if ( style == wxDEFAULT )
-        style = wxNORMAL;
-    if ( weight == wxDEFAULT )
-        weight = wxNORMAL;
+    if ( family == wxFONTFAMILY_DEFAULT )
+        family = wxFONTFAMILY_SWISS;
     if ( size == wxDEFAULT )
         size = 12;
 
@@ -294,21 +290,21 @@ void wxFontMgrFontRefData::SetPointSize(
     m_fontValid = false;
 }
 
-void wxFontMgrFontRefData::SetFamily(int family)
+void wxFontMgrFontRefData::SetFamily(wxFontFamily family)
 {
-    m_info.family = (wxFontFamily)family;
+    m_info.family = family;
     m_fontValid = false;
 }
 
-void wxFontMgrFontRefData::SetStyle(int style)
+void wxFontMgrFontRefData::SetStyle(wxFontStyle style)
 {
-    m_info.style = (wxFontStyle)style;
+    m_info.style = style;
     m_fontValid = false;
 }
 
-void wxFontMgrFontRefData::SetWeight(int weight)
+void wxFontMgrFontRefData::SetWeight(wxFontWeight weight)
 {
-    m_info.weight = (wxFontWeight)weight;
+    m_info.weight = weight;
     m_fontValid = false;
 }
 
